src/coreclr/jit/gentree.h

```diff
@@ -7992,29 +8139,14 @@ inline bool GenTree::OperIsInitBlkOp()
     {
         src = AsBlk()->Data()->gtSkipReloadOrCopy();
     }
-    return src->OperIsInitVal() || src->OperIsConst();
+    return src->OperIsInitVal() || (src->OperIsConst() && !src->IsCnsVec());
```

Comment:
You should change this to 
```suggestion
    return src->OperIsInitVal() || src->IsIntegralConst();
```
InitBlks can only have an integer constant of zero on the RHS.

(I understand this is pre-existing, but let's not make the code more confusing)

Reply:
Fixed.